### PR TITLE
Fix error handling in platform tests

### PR DESCRIPTION
**Pull request by jun-rh**

### CODE REVIEW

File: pkg/tnf/test.go [hunk 1]
```diff
@@ -135,15 +135,23 @@ func (t *Test) RunAndValidateWithFailureCallback(cb func()) {
 	gomega.Expect(err).To(gomega.BeNil())
 }
 
-// RunWithFailureCallback runs the test, invokes the cb on failure
+// RunWithCallbacks runs the test, invokes the cb on failure/error/success
 // This is useful when the testcase needs to continue whether this test result is success or not
-func (t *Test) RunWithFailureCallback(cb func()) {
+func (t *Test) RunWithCallbacks(scb, fcb func(), ecb func(error)) {
```
Comment (hamadise) [inline pkg/tnf/test.go:140]: scb ? ecb ? what do they stand for ?

Comment (hamadise) [reply pkg/tnf/test.go:140]: success call back ? error call back ?

File: pkg/tnf/test.go [hunk 2]
```diff
@@ -135,15 +135,23 @@ func (t *Test) RunAndValidateWithFailureCallback(cb func()) {
...
+			scb()
+		}
+	case FAILURE:
+		if fcb != nil {
+			fcb()
+		}
+	case ERROR:
+		if ecb != nil {
+			ecb(err)
+		}
```
Comment (hamadise) [inline pkg/tnf/test.go:154]: I think generic error printing if the cb is nil would be useful

File: pkg/tnf/test.go (hunk 1 above)
Comment (jun-rh) [reply pkg/tnf/test.go:140]: yes, i can expand the name if it's not clear

File: pkg/tnf/test.go (hunk 2 above)
Comment (jun-rh) [reply pkg/tnf/test.go:154]: generic case should call RunAndValidate() instead

File: pkg/tnf/test.go (hunk 1 above)
Comment (jun-rh) [reply pkg/tnf/test.go]: updated

**Review by hamadise — APPROVED** [review]

**Review by greyerof — APPROVED** [review]

**Pull request merged**